servers/physics_2d/shape_2d_sw.cpp

```diff
@@ -625,6 +638,43 @@ bool ConvexPolygonShape2DSW::intersect_segment(const Vector2 &p_begin, const Vec
 	return inters; //todo
 }
 
+real_t ConvexPolygonShape2DSW::get_area() const {
+
+	if (point_count < 3) return 0;
+
+	real_t signed_area = 0;
+	for (int i = 0; i < point_count; i++) {
+		Vector2 v1 = points[i].pos;
+		Vector2 v2 = points[(i + 1) % point_count].pos;
+		signed_area += v1.x * v2.y - v2.x * v1.y;
+	}
+	signed_area *= 0.5;
+	return ABS(signed_area);
+}
+
+Vector2 ConvexPolygonShape2DSW::get_shape_center_offset() const {
+
+	if (point_count < 3) return Vector2();
+
+	Vector2 centroid;
+	real_t signed_area = 0.0;
+	real_t a = 0.0;
+
+	for (int i = 0; i < point_count; i++) {
+		const Vector2 &v1 = points[i].pos;
+		const Vector2 &v2 = points[(i + 1) % point_count].pos;
+
+		a = v1.x * v2.y - v2.x * v1.y;
+		signed_area += a;
+		centroid += (v1 + v2) * a;
+	}
+
+	signed_area *= 0.5;
+	centroid /= (6.0 * signed_area);
+
+	return centroid;
+}
+
```

Comment:
@akien-mga it would be better if #30222 and #30251 could be merged and used here directly as `Geometry` methods instead of "hardcoding" here, @raphael10241024 would need to rebase later once merged. 🙂

Reply:
How should I convert points to a Vector<Vector2>?